pkg/ls/link.go

```diff
@@ -267,12 +279,14 @@ func (spec *LinkSpec) GetForeignKeys(entityRoot *lpg.Node) ([][]string, error) {
 	//   0          1         2
 	// fk0_key0  fk0_key1  fk0_key2  --> foreign key 1
 	// fk1_key0  fk1_key1  fk1_key2  --> foreign key 2
-	fks := make([][]string, numKeys)
+	// fks := make([][]string, numKeys)
 	for i := 0; i < numKeys; i++ {
 		for key := 0; key < len(spec.FK); key++ {
 			v, _ := GetRawNodeValue(foreignKeyNodes[i][key])
-			fks[i] = append(fks[i], v)
+			if len(foreignKeyInfo) > 0 {
```

Comment:
Create all foreign key info elements here. Instead of 
fks:=make([][]string,numKeys)
do
foreignKeyInfo:=make([]ForeignKeyInfo,numKeys)
and foreignKeyNodes[i][key] is the doc node, its value is the foreign key value